api/v1alpha1/api.go

```diff
@@ -106,6 +106,12 @@ type LLMBackendSpec struct {
 	//
 	// +kubebuilder:validation:Required
 	BackendRef egv1a1.BackendRef `json:"backendRef"`
+
+	// BackendSecurityPolicyRef is the name of the BackendSecurityPolicy resources this backend
+	// is being attached to.
+	//
+	// +optional
+	BackendSecurityPolicyRef *gwapiv1.LocalObjectReference `json:"backendSecurityPolicyRef,omitempty"`
```

Comment:
can you add a CEL validation to ensure that both `apiKey` and `cloudProviderCredentials` won't be defined at the same time